src/AlgoliaSearchCore.js

```diff
@@ -486,9 +514,9 @@ AlgoliaSearchCore.prototype._computeRequestHeaders = function(additionalUA, with
     requestHeaders['x-algolia-tagfilters'] = this.securityTags;
   }
 
-  if (this.extraHeaders) {
-    forEach(this.extraHeaders, function addToRequestHeaders(header) {
-      requestHeaders[header.name] = header.value;
+  if (!isEmpty(this.extraHeaders)) {
```

Comment:
In the end, we do not even need to check isEmpty or not, since the forEach will not iterate on it if it has no properties, let's remove this code?